libs/ui/icon/helm/src/lib/hlm-icon.directive.ts

```diff
@@ -1,12 +1,7 @@
 import { Directive, computed, input } from '@angular/core';
 import { injectHlmIconConfig } from './hlm-icon.token';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const DEFINED_SIZES = ['xs', 'sm', 'base', 'lg', 'xl', 'none'] as const;
-
-type DefinedSizes = (typeof DEFINED_SIZES)[number];
-
-export type IconSize = DefinedSizes | (Record<never, never> & string);
+export type IconSize = 'xs' | 'sm' | 'base' | 'lg' | 'xl' | 'none' | ({} & string);
```

Comment:
I can't find a clear/objective reason for this, but using `Record<never, never>` is recommended over `{}`.

- This is the most "complete" request I could find: https://github.com/dotansimha/graphql-code-generator/issues/6068
- And this is also from the above issue: https://github.com/microsoft/TypeScript/wiki/FAQ#why-are-all-types-assignable-to-empty-interfaces

```suggestion
export type IconSize = 'xs' | 'sm' | 'base' | 'lg' | 'xl' | 'none' | (Record<never, never> & string);
```

Reply:
Yes, typically that is absolutely the case if you are using it as a type, in this case it doesn't change the type, it's a bit of a hack to get the typescript autocomplete to work nicer.

If we do something like 'sm' | 'xs' | string, typescript collapses the union down to just string and we get no autocomplete as you can provide suggestions for type string. This little addition doesn't affect the type, but does stop the union from collapsing giving us autocomplete suggestions 😊

Reply:
Yeah, no, I get that part. I meant that using `Record<never, never>` works in the same way as `{}` to enable the autocomplete, but it is generally more type-safe. Having said that, I can't find anything to cite this 😅

From what I can find, the type-safety issue arises only if we use them standalone, not in union. In this case, there appears to be no difference in the expected output, so this is completely fine.

Reply:
Ah apologies, I see what you mean. Correct, yes, we & the empty object type with a string which avoids the type issues normally associated with {} if used standalone 😊